### PR TITLE
Bug 1882723: vSphere: Create warning on invalid specs and remove defaults

### PR DESCRIPTION
**Pull request by alexander-demichev**
If the user forgets to specify CPU, memory, or disk size we default it to 0 and the reconciler then defaults it to a minimal value.
This PR:
1. Make webhook return warning on missing VM specs
2. Removes defaulting from the controller and let's fail to pass.

### CODE REVIEW

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (JoelSpeed) [inline pkg/apis/machine/v1beta1/machine_webhook.go:995]: IIRC there's something in the controller that caters for this scenario, can we check that before we add this one? 

We want to verify that placing a 0 value in this field will result in a failed machine, have you tried this for each of the three errors you're adding?

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (JoelSpeed) [inline pkg/apis/machine/v1beta1/machine_webhook.go:984]: Why are we removing `the minimum value will be used instead`?

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (JoelSpeed) [inline pkg/apis/machine/v1beta1/machine_webhook.go:987]: Why are we removing `nodes may not boot correctly`?

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (alexander-demichev) [reply pkg/apis/machine/v1beta1/machine_webhook.go:984]: I completely forgot about this behavior.

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (JoelSpeed) [inline pkg/apis/machine/v1beta1/machine_webhook.go:983]: Don't think we actually need the or statement here, 0 < minVSphereCPU is true anyway. Similar goes for the cases below

**Review by JoelSpeed — COMMENTED** [review]
Have you tried this out on a real cluster, what happens if you set CPU 0?

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (michaelgugino) [inline pkg/apis/machine/v1beta1/machine_webhook.go:984]: Are we actually setting the minimum value?  Below it looks like that logic was removed from the reconciler.

**Review by michaelgugino — CHANGES_REQUESTED** [review]

**Review by elmiko — APPROVED** [review]
/lgtm

**Comment by elmiko** [comment]
Mike's comment came in while i was reviewing, removing my lgtm to ensure we don't merge before it's answered.
/lgtm cancel

File: pkg/apis/machine/v1beta1/machine_webhook.go
Comment (JoelSpeed) [reply pkg/apis/machine/v1beta1/machine_webhook.go]: The logic was removed from the webhook, but is being done in the reconcile loop currently
https://github.com/openshift/machine-api-operator/blob/44f29cc5bc7665afe4136103836a018760952839/pkg/controller/vsphere/reconciler.go#L550-L561

**Comment by alexander-demichev** [comment]
@elmiko @michaelgugino Can I get another review?

**Review by elmiko — APPROVED** [review]
/lgtm

**Pull request merged**
